src/settings/settings.go

```diff
@@ -78,6 +78,9 @@ type Settings struct {
 	XdsClientBackoffRandomFactor    float64       `envconfig:"XDS_CLIENT_BACKOFF_RANDOM_FACTOR" default:"0.5"`
 	XdsClientBackoffJitter          bool          `envconfig:"XDS_CLIENT_BACKOFF_JITTER" default:"true"`
 
+	// xDS gRPC client options
+	XdsGrpcClientOptionsMaxMsgSizeInBytes int `envconfig:"XDS_GRPC_CLIENT_OPTIONS_MAX_MSG_SIZE_IN_BYTES" default:""`
```

Comment:
1. the previous env vars dont have the `grpc` prefix
2. should we separate send and receive ? do we even to add a knob for send, its likely to be small

Reply:
@arkodg thanks for your review.
1. changed the config as `XDS_CLIENT_GRPC_OPTIONS_MAX_MSG_SIZE_IN_BYTES `
2. Yes, since [discoveryrequest](https://www.envoyproxy.io/docs/envoy/latest/api-v3/service/discovery/v3/discovery.proto#envoy-v3-api-msg-service-discovery-v3-discoveryrequest) is smaller removed the send message size.

Reply:
should this be `XDS_CLIENT_MAX_MSG_SIZE_IN_BYTES`, since the previous grpc options didnt contain the `grpc` prefix ?

Reply:
Updated via commit: https://github.com/envoyproxy/ratelimit/pull/797/commits/39135a7bc9daa4779e08702ee59939dec4480f18

Reply:
looks good, thanks